src/ParallelAlgorithms/LinearSolver/AsynchronousSolvers/ElementActions.hpp

```diff
@@ -196,6 +196,21 @@ struct PrepareSolve {
     contribute_to_residual_observation<OptionsGroup, ParallelComponent>(
         iteration_id, residual_magnitude_square, cache, array_index);
 
+    // Make the solve the identity operation if no steps are going to be
```

Comment:
Please make this handled by a new flag and not the number of iterations.  I would absolutely not expect this behavior